crates/re_analytics/src/native/mod.rs

```diff
@@ -0,0 +1,28 @@
+mod config;
+mod pipeline;
+mod sink;
+
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+
+pub use config::{Config, ConfigError};
+pub use pipeline::{Pipeline, PipelineError};
+
+#[derive(Default, Clone)]
+pub(crate) struct AbortSignal {
+    aborted: Arc<AtomicBool>,
+}
+
+impl AbortSignal {
+    pub(crate) fn new() -> Self {
+        Self::default()
+    }
+
+    pub(crate) fn signal(&self) {
```

Comment:
I think this is a bit more readable, otherwise `return abort_signal.signal();` later looks like you are returning the signal contained in the `abort_signal` somehow

```suggestion
    pub(crate) fn abort(&self) {
```